src/crawlee/abstract_http_crawler/_abstract_http_parser.py

```diff
@@ -0,0 +1,93 @@
+from __future__ import annotations
+
+from abc import ABC, abstractmethod
+from dataclasses import dataclass
+from typing import TYPE_CHECKING, Generic
+
+from crawlee._utils.blocked import RETRY_CSS_SELECTORS
+from crawlee._utils.docs import docs_group
+from crawlee.abstract_http_crawler._http_crawling_context import TParseResult
+
+if TYPE_CHECKING:
+    from collections.abc import Iterable
+
+    from crawlee.http_clients import HttpResponse
+
+
+@docs_group('Classes')
+@dataclass(frozen=True)
+class BlockedInfo:
+    """Information about whether the crawling is blocked. If reason is empty, then it means it is not blocked."""
+
+    reason: str
+
+    def __bool__(self) -> bool:
+        """No reason means no blocking."""
+        return bool(self.reason)
```

Comment:
Just wondering whether this is a good location for `BlockedInfo`? Couldn't we utilize it in the `PlaywrightCrawler` as well? Maybe it should be located on the basic crawler / basic crawling context level. What do you think?

Reply:
Understand, but placing it at the basic crawler / basic crawling context level now could help us avoid introducing a breaking change in the future if `PlaywrightCrawler` starts using it.

Reply:
Ok